src/current/_includes/molt/migration-prepare-database.md

```diff
@@ -1,15 +1,141 @@
+#### Create migration user on source database
+
+Create a dedicated migration user (e.g., `MIGRATION_USER`) on the source database. This user is responsible for reading data from source tables during the migration.
+
 <section class="filter-content" markdown="1" data-scope="postgres">
-Ensure that the PostgreSQL database is configured for replication. Enable logical replication by setting `wal_level` to `logical` in `postgresql.conf` or in the SQL shell. For example:
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE USER MIGRATION_USER WITH PASSWORD 'password';
+~~~
+
+Grant the user privileges to connect, view schema objects, and select the tables you migrate.
 
 {% include_cached copy-clipboard.html %}
 ~~~ sql
-ALTER SYSTEM SET wal_level = 'logical';
+GRANT CONNECT, RESOURCE ON DATABASE source_database TO MIGRATION_USER;
+GRANT USAGE ON SCHEMA migration_schema TO MIGRATION_USER;
+GRANT SELECT ON ALL TABLES IN SCHEMA migration_schema TO MIGRATION_USER;
+ALTER DEFAULT PRIVILEGES IN SCHEMA migration_schema GRANT SELECT ON TABLES TO MIGRATION_USER;
 ~~~
 </section>
 
 <section class="filter-content" markdown="1" data-scope="mysql">
-Ensure that the MySQL database is configured for replication.
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE USER MIGRATION_USER@'%' IDENTIFIED BY 'password';
+~~~
+
+Grant the user privileges to select only the tables you migrate:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+GRANT SELECT ON source_database.* TO MIGRATION_USER@'%';
+FLUSH PRIVILEGES;
+~~~
+</section>
+
+<section class="filter-content" markdown="1" data-scope="oracle">
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE USER MIGRATION_USER IDENTIFIED BY '{password}';
+~~~
+
+{{site.data.alerts.callout_info}}
+When migrating from Oracle Multitenant (PDB/CDB), this should be a [common user](https://docs.oracle.com/database/121/ADMQS/GUID-DA54EBE5-43EF-4B09-B8CC-FAABA335FBB8.htm). Prefix the username with `C##` (e.g., `C##MIGRATION_USER`).
+{{site.data.alerts.end}}
+
+Grant the migration user privileges to connect, read metadata, and `SELECT` and `FLASHBACK` the tables you plan to migrate. The tables should all reside in a single schema (e.g., `migration_schema`). For details, refer to [Schema and table filtering](#schema-and-table-filtering).
```

Comment:
@noelcrl is "schema" the precise name for this concept in Oracle? Would it also help here to link Oracle docs related to `SELECT` and `FLASHBACK` grants or show a sample `GRANT` query?